files/system/usr/share/ublue-os/just/60-custom.just

```diff
@@ -56,10 +56,21 @@ remove-kargs-hardening:
       --delete-if-present="debugfs=off"
     echo "Hardening kargs removed."
 
-# Harden flatpaks by preloading hardened_malloc
+# Harden flatpaks by preloading hardened_malloc (highest supported hwcap)
 harden-flatpak:
     #!/usr/bin/bash
-    flatpak override --user --filesystem=host-os:ro --env=LD_PRELOAD=/var/run/host/usr/lib64/libhardened_malloc.so
+    flatpak override --user --filesystem=host-os:ro
+#   `ld-linux-x86-64.so.2 --help` prints ld.so linker info, including detected hwcaps support. Grep those, then use cut and substring selection (:0:1) to isolate the 1st character after 'v', which will be 4, 3, or 2, corresponding to the system's highest supported hwcap. On x86_64-v1 systems, grep finds no matches, leaving our variables empty.
+    uarches="$(/var/run/host/usr/lib64/ld-linux-x86-64.so.2 --help | grep '(supported, searched)' | cut -d'v' -f2)"
```

Comment:
this doesn't work properly:

bash: /var/run/host/usr/lib64/ld-linux-x86-64.so.2: No such file or directory

Reply:
please provide evidence of your testing

Reply:
/var/run/host is only defined for flatpaks I believe. Were you testing from within a flatpak shell? :smile:

Reply:
Not that I'm aware of. But I was using the "dx" edition of bluefin, which apparently adds copies of those libraries in /var/run/host/usr/lib64/ (or makes that directory available to non-flatpak apps). I've now switched to silverblue and see that the libraries are only in /usr/lib64/ . I've updated my fork to point request it from /usr/lib64/ , so the code should now work on non-"dx" images.

Reply:
they probably symlink it